### ISSUE TITLE
`suppressible-exception` (`SIM105`) false negative when multiple lines in `try` block

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8593]
```py
try: # error
    ...
except Exception:
    pass

try: # no error
    ...
    ...
except Exception:
    pass
```
[playground](https://play.ruff.rs/0ae2bc17-7268-450b-b602-298728301078)

**Comment by KotlinIsland** [comment]
I think this is intentional, although I don't really see why.

**Comment by konstin** [comment]
For context, this is due to #1947/#1948 and flake8-simplify doing the same (https://github.com/MartinThoma/flake8-simplify/blob/60e0ddbc82973af673259850cc00c6e8221f8643/flake8_simplify/rules/ast_try.py#L46). I'm not sure if that's the best way to handle this though

**Comment by charliermarsh** [comment]
I think the current restriction is not unreasonable, but I'd be open to reviewing a PR that modifies the rule to use the more limited exemption from #1947:

> If a `try:` block has an unreachable end because all its non-exceptional control flow paths end with a `return`, `raise`, `break`, or `continue`, should we exempt it from SIM105?

Though it'd be contingent on reviewing the changes in the ecosystem checks.

**Comment by DetachHead** [comment]
workaround: use [`S110`](https://docs.astral.sh/ruff/rules/try-except-pass/) instead. though it has a different error message, it seems to be pretty much the same rule except it also covers this case as well.

perhaps `SIM105` should be deprecated in favor of `S110` and `S110`'s error message can be updated to also mention `contextlib.suppress`?

**Comment by chirizxc** [comment]
Can we remove the restriction for certain circumstances?

For example, if there are only imports inside the block?
```python
try:
    from pydantic import ConfigDict, TypeAdapter, ValidationError
    from pydantic.main import IncEx
except ImportError:
    pass

**Comment by chirizxc** [comment]
or make `detect-multiline` parameter and by default it will be false

```toml
[lint.flake8-simplify]
detect-multiline = true
```

**Comment by DetachHead** [comment]
why would it make a difference if the block only contains import statements?

**Comment by chirizxc** [comment]
> why would it make a difference if the block only contains import statements?

The code is just an example, I suggest adding:
```toml
[lint.flake8-simplify]
detect-multiline = true
```
so that it can detect not only single lines.

**Comment by DetachHead** [comment]
i still don't understand why anybody would only want the error to be reported on single-line `try`/`except` statements, so i don't see the point making it an option
